File: tests/integration/initialize-project/projects/python-fastapi/app/services/user_service.py
"""User service business logic."""
from typing import List, Optional
from passlib.context import CryptContext

from app.schemas.user import UserCreate, UserResponse

pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")


class UserService:
    """User service for managing users."""

    def __init__(self):
        self.users: List[dict] = []
        self.next_id = 1

    async def get_all(self) -> List[UserResponse]:
        """Get all users."""
        return [UserResponse(**user) for user in self.users]

    async def get_by_id(self, user_id: int) -> Optional[UserResponse]:
        """Get user by ID."""
        user = next((u for u in self.users if u["id"] == user_id), None)
        return UserResponse(**user) if user else None

    async def create(self, user_data: UserCreate) -> UserResponse:
        """Create a new user."""
        hashed_password = pwd_context.hash(user_data.password)
        user = {
            "id": self.next_id,
            "email": user_data.email,
            "full_name": user_data.full_name,
            "hashed_password": hashed_password,
        }
        self.users.append(user)
        self.next_id += 1
        return UserResponse(**user)

File: tests/integration/initialize-project/projects/simple-api/src/services/user.service.ts
import bcrypt from 'bcrypt';

interface User {
  id: string;
  email: string;
  password: string;
  name: string;
}

export class UserService {
  private users: User[] = [];

  async findAll(): Promise<Omit<User, 'password'>[]> {
    return this.users.map(({ password, ...user }) => user);
  }

  async findById(id: string): Promise<Omit<User, 'password'> | null> {
    const user = this.users.find(u => u.id === id);
    if (!user) return null;
    const { password, ...userWithoutPassword } = user;
    return userWithoutPassword;
  }

  async create(userData: Omit<User, 'id'>): Promise<Omit<User, 'password'>> {
    const hashedPassword = await bcrypt.hash(userData.password, 10);
    const user: User = {
      id: Math.random().toString(36).substr(2, 9),
      ...userData,
      password: hashedPassword
    };
    this.users.push(user);
    const { password, ...userWithoutPassword } = user;
    return userWithoutPassword;
  }
}
